darwin/dataset/identifier.py

```diff
@@ -22,18 +26,19 @@ def __str__(self):
         return output
 
 
-def parse(slug: str):
-    try:
-        if "/" in slug:
-            team, slug = slug.split("/")
-        else:
-            team = None
-
-        if ":" in slug:
-            dataset, version = slug.split(":")
-        else:
-            dataset = slug
-            version = None
-        return team, dataset, version
-    except ValueError as e:
-        raise type(e)(f"Invalid dataset identifier {slug}")
+def _parse(slug: str) -> Tuple[str, str, Optional[str]]:
+    version: Optional[str] = None
+
+    if not _is_slug_valid(slug):
+        raise ValueError(f"Invalid dataset identifier {slug}")
+
+    team, dataset = slug.split("/")
+
+    if ":" in dataset:
+        dataset, version = dataset.split(":")
+
+    return team, dataset, version
+
+
+def _is_slug_valid(slug: str) -> bool:
+    return re.fullmatch(r"[\w\-]+/[\w\-]+(:[\w\.]+)?", slug) is not None
```

Comment:
I think this blocks us from doing just "dataset name" which we could before! 

```suggestion
    return re.fullmatch(r"([\w\-]+/)?[\w\-]+(:[\w\.]+)?", slug) is not None
```